src/ngraph_builder.cc

```diff
@@ -4187,14 +4187,63 @@ static Status TranslateSelectOp(
   TF_RETURN_IF_ERROR(
       GetInputNodes(ng_op_map, op, &ng_input1, &ng_input2, &ng_input3));
 
-  // broadcast to make all tensors same shape, as required by ngraph select op
-  std::tie(ng_input1, ng_input2) =
-      ng::builder::numpy_broadcast(std::make_pair(ng_input1, ng_input2));
-  std::tie(ng_input2, ng_input3) =
-      ng::builder::numpy_broadcast(std::make_pair(ng_input2, ng_input3));
+  if (ng_input2->get_shape() != ng_input2->get_shape()) {
+    return errors::InvalidArgument(
+        "Input tensors 2 and 3 should have same shape");
+  }
+
+  auto ng_input1_shape = ng_input1->get_shape();
+  auto ng_input2_shape = ng_input2->get_shape();
+
+  auto ng_input1_size = ng_input1->get_shape().size();
+  auto ng_input2_size = ng_input2->get_shape().size();
+
+  int length;
+  shared_ptr<ng::Node> ng_input_new, ng_select;
+  ng::AxisVector ng_axis_order;
+
+  for (size_t i = 0; i < ng_input2_size; i++) {
+    if (ng_input1_shape[0] == ng_input2_shape[i]) {
+      length = ng_input2_size - i - ng_input1_size;
+    }
+  }
 
-  auto ng_select = ConstructNgNode<ng::op::Select>(op->name(), ng_input1,
-                                                   ng_input2, ng_input3);
+  if (length != 0) {
+    // Condition tensor will be modified to align the condition tensor
+    // shape with input tensor shape index and fill the rest of the vector with
+    // 1s
+    // Eg: consition tensor [7], input tensor [7, 3, 2, 1]
+    // After Reshape, condition tensor will be [7, 1 ,1 ,1] for auto broadcast.
+
+    std::vector<size_t> tmp_vector((length + ng_input1_size), 1);
+    tmp_vector[0] = ng_input1_shape[0];
+    for (size_t i = 0; i < ng_input1_size; i++) {
+      ng_axis_order.push_back(i);
+    }
+
+    ng_input_new = ConstructNgNode<ng::op::Reshape>(op->name(), ng_input1,
+                                                    ng_axis_order, tmp_vector);
+
+    // broadcast to make all tensors same shape, as required by ngraph select op
+    std::tie(ng_input_new, ng_input2) =
+        ng::builder::numpy_broadcast(std::make_pair(ng_input_new, ng_input2));
+    std::tie(ng_input2, ng_input3) =
+        ng::builder::numpy_broadcast(std::make_pair(ng_input2, ng_input3));
+
+    ng_select = ConstructNgNode<ng::op::Select>(op->name(), ng_input_new,
+                                                ng_input2, ng_input3);
```

Comment:
the broadcast and select op can move outside the `if`

simply: `auto broadcast_input_1 = length != 0 ? ng_input_new : ng_input1` or something like that